lib/libc/mips/gen/_setjmp.S

```diff
@@ -115,29 +115,21 @@ LEAF(_setjmp)
 	INT_S	v0, (_JB_FPREG_FCSR * SZREG)(a0)
 #endif /* ! __mips_soft_float */
 #if defined(__CHERI__)
-	/* jmp_buf is 64-bit aligned, so we need to round it up. */
```

Comment:
For completeness sake, we should make all of these functions fail if they encounter an under-aligned jmp_buf.

Reply:
Well, the `csc`/`clc` will trap if unaligned, and `setjmp`/`longjmp` can never fail (other than trapping).

Reply:
It's not pointless; it returns 0 the first time, and subsequent returns are whatever was passed to the prior `longjmp` call (or 1 if 0 was passed to avoid ambiguity). We could indeed use the BSD-specific `longjmperror` for this, though personally I'd say it's marginally easier to debug if you just get the unaligned trap, especially given that's the only option for `setjmp`.

Reply:
Oh right, the FreeBSD manpage needlessly confusing and states:
```
     The sigsetjmp(), setjmp(), and _setjmp() functions save their calling
     environment in env.  Each of these functions returns 0.
```